yast2_rdp.pm

```diff
@@ -0,0 +1,194 @@
+        wait_screen_change { send_key "alt-f" };
+        wait_still_screen;
+    } elsif ($label eq 'Firewall') {
+        assert_screen("$tag");
+        wait_screen_change { send_key "alt-o" };
+        wait_screen_change { send_key "alt-f" };
+        wait_still_screen;
+    } else {
+        die "$label is a unknown option in check_box";
+    }
+}
+
+sub run {
+    my $self = shift;
+    select_console("x11");
+
+    # work before test yast2-rdp
+    setup_system;
+
+    # yast2-rdp starts xrdp service. At this time, xrdp is NOT installed probably
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    if (match_has_tag('yast2-rdp-ui-install-xrdp')) {
+        assert_and_click('yast2-rdp-ui-install-xrdp');
+        wait_screen_change { send_key "alt-i" };
+        wait_still_screen;
+    }
+    check_box('RDP', 'yast2-rdp-initial-after-installing-xrdp');
+
+    # check if xrdp service starts.
+    my $start  = check_service('General', 'echo $(systemctl is-active xrdp.service)');
+    my $enable = check_service('General', 'echo $(systemctl is-enabled xrdp.service)');
+    if (!($start and $enable)) {
+        die "It failed that yast2-rdp starts xrdp service";
+    }
+
+    # yast2-rdp stops xrdp service.
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    check_box('RDP', 'yast2-rdp-ui-enable-xrdp');
+
+    # check if xrdp service stops.
+    $start  = check_service('General', 'echo $(systemctl is-active xrdp.service)');
+    $enable = check_service('General', 'echo $(systemctl is-enabled xrdp.service)');
+    if ($start or $enable) {
+        die "It failed that yast2-rdp stops xrdp service";
+    }
+
+    # yast2-rdp adds "ms-wbt" service into these zones "external home internal public trusted work"
+    # openning port in firewall.
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    check_box('Firewall', 'yast2-rdp-initial-after-installing-xrdp');
+
+    # check if openning port in firewall correctly.
+    $start = check_service('General', 'echo $(systemctl is-active firewalld.service)');
+    if ($start) {
+        my $service = check_service('Firewall', 'echo $(firewall-cmd --list-services | grep -Eo "ms-wbt")X');
+        if (!$service) {
+            die "It failed that yast2-rdp opens port in firewall";
+        }
+    } else {
+        my $service = check_service('Firewall', 'echo $(firewall-offline-cmd --list-services | grep -Eo "ms-wbt")X');
+        if (!$service) {
+            die "It failed that yast2-rdp opens port in firewall";
+        }
+    }
+
+    # yast2-rdp deletes "ms-wbt" service from these zones "external home internal public trusted work"
+    # closing port in firewall.
+    $self->launch_yast2_module_x11("rdp", match_timeout => 120);
+    check_box('Firewall', 'yast2-rdp-ui-open-port-in-firewall');
+
+    # check if closing port in firewall correctly.
+    $start = check_service('General', 'echo $(systemctl is-active firewalld.service)');
```

Comment:
on SLE12 is SuSEFirewall2 .... , on SLE15+ , and Leap15+ firewalld

Reply:
yes, thanks for your advice.